t/lib/OpenQA/Test/Utils.pm

```diff
@@ -395,16 +407,28 @@ sub c_worker {
                 handle_command => sub {
                     my ($self, $tx, $json) = @_;
                     log_debug('Received ws message: ' . Dumper($json));
+
+                    # if we've got a single job ID and a rejection reason simulate a worker
+                    # which rejects the job
+                    my $rejection_reason = $options{rejection_reason};
+                    return undef unless defined $rejection_reason;
+                    my $job_id = (($json // {})->{job} // {})->{id};
```

Comment:
Would autovivification be harmuful here?

Reply:
I guess I've forgotten about that super cool Perl feature when writing this code. I suppose it would actually be quite useful here :-)